src/rpc/rawtransaction.cpp

```diff
@@ -252,6 +298,49 @@ static UniValue getrawtransaction(const JSONRPCRequest& request)
     UniValue result(UniValue::VOBJ);
     if (blockindex) result.pushKV("in_active_chain", in_active_chain);
     TxToJSON(*tx, hash_block, txType, txStatus,  result);
+
+    if (fVerbose == 2) {
+        int64_t sum_vin = 0;
+        UniValue vins_info(UniValue::VARR) ;
+        for (auto& vin : tx->vin) {
+            CBaseTransactionRef prev_tx;
+            uint256 prev_hash_block;
+            vaulttxnstatus prev_txStatus;
+            if (GetTransaction(vin.prevout.hash, prev_tx, Params().GetConsensus(), prev_hash_block, blockindex, &prev_txStatus)) {
+                sum_vin += prev_tx->vout[vin.prevout.n].nValue;
+                txnouttype type;
+                std::vector<CTxDestination> addresses;
+                int nRequired;
+                if (ExtractDestinations(prev_tx->vout[vin.prevout.n].scriptPubKey, type, addresses, nRequired)) {
+                    UniValue res(UniValue::VOBJ);
+                    res.pushKV("txid", prev_tx->GetHash().GetHex());
+                    res.pushKV("amount", ValueFromAmount(prev_tx->vout[vin.prevout.n].nValue));
+                    res.pushKV("vout", static_cast<uint64_t>(vin.prevout.n));
+
+                    if (addresses.size() == 1) {
+                        res.pushKV("address", EncodeDestination(addresses[0]));
+                    } else {
+                        UniValue a(UniValue::VARR);
+                        for (const CTxDestination& addr : addresses) {
+                            a.push_back(EncodeDestination(addr));
+                        }
+                        res.pushKV("address", a);
+                    }
+                    vins_info.push_back(std::move(res));
+                }
+            }
+        }
+
+        int64_t sum_vout = 0;
+        for (auto& vout : tx->vout) {
+            sum_vout += vout.nValue;
+        }
+
+        result.pushKV("fee", ValueFromAmount(sum_vin - sum_vout));
```

Comment:
doesn't it require special handling for coinbase transactions? (will result in fee < 0
sth around this lines?
`sum_vin == 0 ? 0 : sum_vin - sum_vout`

Reply:
Nope, its not point of interest.

Reply:
Coinbase case should be covered here.